common/ferrostar/src/routing_adapters/osrm/models.rs

```diff
@@ -185,6 +187,23 @@ pub struct BannerContent {
     /// the original travel.
     #[serde(rename = "degrees")]
     pub roundabout_exit_degrees: Option<u16>,
+    /// Extra information for displaying the instructions (ex lanes, images, etc)
+    pub components: Vec<BannerContentComponent>,
+}
+
+/// Details used to display extra information for the banner instructions.
+/// Note that while all of these are parsed, not all of them are currently
+/// in use.
+#[derive(Deserialize, Debug)]
+pub struct BannerContentComponent {
+    #[serde(rename = "type")]
+    pub component_type: Option<String>,
+    pub directions: Option<Vec<String>>,
+    pub active: Option<bool>,
+    pub image_base_url: Option<String>,
+    pub abbr: Option<String>,
+    pub abbr_priority: Option<u8>,
+    pub active_direction: Option<String>,
```

Comment:
Correct; intersections also have detailed lane info, which is what I originally thought this PR was about 😂 My current understanding (not having designed either API) is that these are complementary, but that the Mapbox usage is the more common one across navigation SDKs. The detailed intersection info may be useful in specific contexts though, so we modeled it just in case.